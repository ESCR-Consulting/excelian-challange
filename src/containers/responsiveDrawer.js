import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Drawer from 'material-ui/Drawer';
import Hidden from 'material-ui/Hidden';
import List, {
    ListItem,
    ListItemIcon,
    ListItemText,
} from 'material-ui/List';
import AccountCircleIcon from 'material-ui-icons/AccountCircle';
import StarIcon from 'material-ui-icons/Star';
import VerifiedUserIcon from 'material-ui-icons/VerifiedUser';
import PeopleIcon from 'material-ui-icons/People';
import InboxIcon from 'material-ui-icons/Inbox';
import MenuAppBar from '../components/MenuAppBar';
import UserList from '../containers/userList';

const drawerWidth = 240;

const styles = theme => ({
    root: {
        width: '100%',
        height: '100vh',
        zIndex: 1,
        overflow: 'hidden',
    },
    appFrame: {
        position: 'relative',
        display: 'flex',
        width: '100%',
        height: '100%',
    },
    appBar: {
        position: 'absolute',
        marginLeft: drawerWidth,
        [theme.breakpoints.up('sm')]: {
            width: `calc(100% - ${drawerWidth}px)`,
        },
    },
    toolBarHeader: {
        maxWidth: drawerWidth,
        position: 'relative',
        [theme.breakpoints.down('sm')]: {
            maxWidth: '100%',
        },
    },
    navIconHide: {
        [theme.breakpoints.up('sm')]: {
            display: 'none',
        },
    },
    drawerHeader: theme.mixins.toolbar,
    drawerPaper: {
        width: 250,
        [theme.breakpoints.up('sm')]: {
            width: drawerWidth,
            position: 'relative',
            height: '100%',
        },
    },
    content: {
        backgroundColor: theme.palette.background.default,
        width: '100%',
    },
});

class ResponsiveDrawer extends React.Component {
    state = {
        mobileOpen: true,
    };

    handleDrawerToggle = () => {
        this.setState({ mobileOpen: !this.state.mobileOpen });
    };

    render() {
        const { classes, theme } = this.props;
        const drawer = (
            <div>
                <AppBar
                    color="default"
                    className={classes.toolBarHeader}
                >
                    <Toolbar className={classes.toolBarHeader}>
                        <Typography type="title" color="inherit">
                            Users
                        </Typography>
                    </Toolbar>
                </AppBar>
                <List>
                    <ListItem button>
                        <ListItemIcon>
                            <AccountCircleIcon />
                        </ListItemIcon>
                        <ListItemText primary="All Users" />
                    </ListItem>
                    <ListItem button>
                        <ListItemIcon>
                            <StarIcon />
                        </ListItemIcon>
                        <ListItemText primary="Favourites" />
                    </ListItem>
                    <ListItem button>
                        <ListItemIcon>
                            <VerifiedUserIcon />
                        </ListItemIcon>
                        <ListItemText primary="Administrators" />
                    </ListItem>
                    <ListItem button>
                        <ListItemIcon>
                            <PeopleIcon />
                        </ListItemIcon>
                        <ListItemText primary="Non-Admins" />
                    </ListItem>
                    <ListItem button>
                        <ListItemIcon>
                            <InboxIcon />
                        </ListItemIcon>
                        <ListItemText primary="Archived" />
                    </ListItem>
                </List>
            </div>
        );

        return (
            <div className={classes.root}>
                <MenuAppBar />
                <div className={classes.appFrame}>
                    <Hidden smUp>
                        <Drawer
                            type="temporary"
                            anchor={theme.direction === 'rtl' ? 'right' : 'left'}
                            open={this.state.mobileOpen}
                            classes={{
                                paper: classes.drawerPaper,
                            }}
                            onRequestClose={this.handleDrawerToggle}
                            ModalProps={{
                                keepMounted: true, // Better open performance on mobile.
                            }}
                        >
                            {drawer}
                        </Drawer>
                    </Hidden>
                    <Hidden smDown implementation="css">
                        <Drawer
                            type="permanent"
                            open
                            classes={{
                                paper: classes.drawerPaper,
                            }}
                        >
                            {drawer}
                        </Drawer>
                    </Hidden>
                    <main className={classes.content}>
                        <AppBar
                            position="static"
                            color="default"
                            style={{
                                position: 'relative'
                            }}>
                            <Toolbar>
                                <Typography type="title" color="inherit">
                                    All Users
                                </Typography>
                            </Toolbar>
                        </AppBar>
                        <UserList />
                    </main>
                </div>
            </div>
        );
    }
}

ResponsiveDrawer.propTypes = {
    classes: PropTypes.object.isRequired,
    theme: PropTypes.object.isRequired,
};

function mapStateToProps({app: {users} }) {
    return {
        users // gives our component access to state through props.users
    }
}

export default connect(mapStateToProps)(withStyles(styles, { withTheme: true })(ResponsiveDrawer));
