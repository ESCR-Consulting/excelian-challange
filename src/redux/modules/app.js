const TOGGLE_DRAWER = 'TOGGLE_DRAWER';

export function handleDrawerToggle(show){
    return {
        type: TOGGLE_DRAWER,
        show
    }
}

const initialState = {
    users: [
        {
            "forename": "Susy",
            "surname": "Cuningham",
            "email": "suzy.cuningham@gmail.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Bobby",
            "surname": "Daniels",
            "email": "bobbyD@outlook.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "John",
            "surname": "Walker",
            "email": "johnnyWalker@blue.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Eddy",
            "surname": "Stevens",
            "email": "eStevens@yahoo.com",
            "lastLogin": "1/10/2015"
        },{
            "forename": "Jan",
            "surname": "Williams",
            "email": "jDubz@msn.com",
            "lastLogin": "1/10/2015"
        }
    ],
    mobileOpen: false
};

export default function reducer(state = initialState, action){
    switch (action.type){
        case TOGGLE_DRAWER:
            return {...state, mobileOpen: !action.show}
        default:
            return state;
    }
}
