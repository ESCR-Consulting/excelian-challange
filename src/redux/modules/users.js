const SEARCH_CHANGE = 'SEARCH_CHANGE';
const SORT_USERS = 'SORT_USERS';

export function handleSearchChange(value){
    return {
        type: SEARCH_CHANGE,
        value
    }
}

export function sortUsers(sortDirection){
    return {
        type: SORT_USERS,
        sortDirection
    }
}

export const userList = [
    {
        "name": "Susy Cuningham",
        "email": "suzy.cuningham@gmail.com",
        "lastLogin": "2017-12-10T00:00:00+00:00"
    },{
        "name": "Bobby Daniels",
        "email": "bobbyD@outlook.com",
        "lastLogin": "2017-11-10T00:00:00+00:00"
    },{
        "name": "John Walker",
        "email": "johnnyWalker@blue.com",
        "lastLogin": "2017-06-10T00:00:00+01:00"
    },{
        "name": "Eddy Stevens",
        "email": "eStevens@yahoo.com",
        "lastLogin": "2016-01-10T00:00:00+00:00"
    },{
        "name": "Jan Williams",
        "email": "jDubz@msn.com",
        "lastLogin": "2015-12-31T00:00:00+00:00"
    }
]
const initialState = {
    userList,
    searchValue: '',
    sortDirection: true,
};

export default function reducer(state = initialState, action){
    switch (action.type){
        case SEARCH_CHANGE:
            return {
                ...state,
                searchValue: action.value,
                userList: action.value !== ""
                    ? state.userList.filter(user => user.name.toLowerCase()
                        .match(action.value.toLowerCase()))
                    : userList
            };
        case SORT_USERS:
            return action.sortDirection
                ? {...state, userList: [...state.userList.sort((a,b) =>
                        a.name < b.name
                            ? -1
                            : a.name > b.name
                            ? 1
                            : 0)],
                sortDirection: !action.sortDirection}
                : {...state, userList: [...state.userList.sort((a,b) =>
                        a.name > b.name
                            ? -1
                            : a.name < b.name
                            ? 1
                            : 0)],
                sortDirection: !action.sortDirection}
        default:
            return state;
    }
}
