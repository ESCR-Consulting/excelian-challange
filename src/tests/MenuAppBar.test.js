import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import MenuAppBar from '../components/MenuAppBar';
import { handleDrawerToggle } from "../redux/modules/responsiveDrawer";

const classes = {
    root: '[object Object]',
    flex: '[object Object]',
    menuButton: '[object Object]',
}
configure({ adapter: new Adapter() });

describe('MenuAppBar', () => {
    it('should render correctly', () => {
        const output = shallow(
            <MenuAppBar
                classes={classes}
                mobileOpen={true}
                handleDrawerToggle={handleDrawerToggle(true)}
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
