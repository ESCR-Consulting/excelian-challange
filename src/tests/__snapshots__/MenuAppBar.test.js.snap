// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuAppBar should render correctly 1`] = `
<MenuAppBar
  classes={
    Object {
      "flex": "MenuAppBar-flex-2 [object Object]",
      "menuButton": "MenuAppBar-menuButton-3 [object Object]",
      "root": "MenuAppBar-root-1 [object Object]",
    }
  }
  handleDrawerToggle={
    Object {
      "show": true,
      "type": "TOGGLE_DRAWER",
    }
  }
  mobileOpen={true}
/>
`;
